ports/esp32/machine_hw_spi.c

```diff
@@ -213,7 +213,7 @@ STATIC void machine_hw_spi_init_internal(
         .clock_speed_hz = self->baudrate,
         .mode = self->phase | (self->polarity << 1),
         .spics_io_num = -1, // No CS pin
-        .queue_size = 1,
+        .queue_size = 2,
```

Comment:
what does this actually do, does it allocate internal memory for both queues?  what about the `len <= 4` case in `machine_hw_spi_transfer` below, is that ok that it still uses 1 queue?

Reply:
This is a good question. The IDF documentation just says:

> Transaction queue size. This sets how many transactions can be ‘in the air’ (queued using `spi_device_queue_trans` but not yet finished using `spi_device_get_trans_result`) at the same time.

I am assuming that it something to do with memory allocation, but I assume the queue is just pointers to the transaction structures, so I can't imagine it being that big a deal.

For small transfers, these still count as a single transaction item in the queue, it's just that the bits are stored straight in the transaction instead of being a pointer. So, yes, small transfers will still use 1 queue item.

Reply:
Ok!